packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -97,6 +97,10 @@ function registerCommands(
     'sfdx.force.source.deploy',
     forceSourceDeploy
   );
+  const forceSourceDeployCurrentFileCmd = vscode.commands.registerCommand(
```

Comment:
The rest of the implementation was checked in as a part of this PR: https://github.com/forcedotcom/salesforcedx-vscode/commit/36f05c627e92267d10238a225129b939bd620a35

I noticed during manual testing that when right-clicking on a file and selecting `SFDX: Deploy This Source File To Org` produced no action because the command was not initialized properly. 
![screen shot 2018-07-30 at 5 11 25 pm](https://user-images.githubusercontent.com/1639375/43428469-a4601268-941b-11e8-8989-9dff41edb25d.png)